File: 16주차/SG/Pascal's_Triangle_II.cpp
#include <bits/stdc++.h>
using namespace std;
int dp[50][50];
class Solution
{
public:
    vector<int> getRow(int rowIndex)
    {
        rowIndex++;
        dp[1][1] = 1;
        vector<int> ans;
        for (int i = 2; i <= rowIndex; i++)
        {
            for (int j = 1; j <= i; j++)
            {
                dp[i][j] = dp[i - 1][j - 1] + dp[i-1][j];
            }
        }
        for (int i = 1; i <= rowIndex; i++)
        {
            ans.push_back(dp[rowIndex][i]);
        }
        return ans;
    }
};

File: 16주차/NH/Pascal's Triangle II.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
//1차원 배열로 구현한 파스칼 삼각형
 function getRow(rowIndex) {
    let row = [1];
     
     for(let i = 1 ; i <= rowIndex ; i++) {
         for(let j = i; j > 0; j--) {
            
             if(j === i)
                 row[j] = 1;
             else
                 row[j] = row[j - 1] + row[j];
         }
     }
     return row;
 };
 //2차원 배열로 구현한 파스칼 삼각형 
 function getRow(rowIndex) {
     let n=new Array(rowIndex+1);
     for(let i=0; i<n.length; i++){
         n[i]=new Array(i+1);
     }
     for(let i=0; i<=rowIndex; i++){
         n[i][0]=1;
         n[i][i]=1;
     }
     for(let j=2; j<=rowIndex; j++){
         for(let k=1; k<j; k++){
             n[j][k]=n[j-1][k-1]+n[j-1][k];                 
         }
     } 
     return n[rowIndex]
 };
 